lib/OpenQA/Utils.pm

```diff
@@ -88,33 +90,54 @@ sub needledir {
     return productdir($distri, $version) . '/needles';
 }
 
-sub needle_info($$$) {
+sub needle_info {
     my $name    = shift;
     my $distri  = shift;
     my $version = shift;
+    my $fn      = shift;
     local $/;
 
     my $needledir = needledir($distri, $version);
 
-    my $fn = "$needledir/$name.json";
+    if (!$fn) {
+        $fn = "$needledir/$name.json";
+    }
+    else {
+        $needledir = dirname($fn);
+    }
+
+    # make sure needledir is a subdir of testcasedir
+    if (index(Cwd::realpath($needledir), Cwd::realpath($testcasedir)) != 0) {
+        warn "$needledir is not a subdir of $testcasedir";
+        return;
+    }
```

Comment:
I don't see why this is needed. It prevents me from using a repo for needles and symlink it into the testcasedir. Why use `Cwd::realpath` here?